### PR TITLE
ENS airdrop + assets

### LINKED ISSUE #3686: Add support for ENS airdrop
## Abstract

The ENS airdrop starts tomorrow. We should add support/detection for it in our upcoming release.

https://twitter.com/LefterisJP/status/1457445408981061639

## Task

Use the json file there but post-process it. Create a new file with:
1. Checksummed addresses
2. full balances (past + future) tokens. So simply add them.
3. Remove all other info to save size.

Then add it to our airdrop detection as a gist like the rest of the airdrops we support.

### PR DESCRIPTION
**Pull request by yabirgb**
Closes #3686

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

File: rotkehlchen/chain/ethereum/airdrops.py [hunk 1]
```diff
@@ -83,6 +84,11 @@
         A_FOX,
         'https://shapeshift.com/shapeshift-decentralize-airdrop',
     ),
+    'ens': (
+        'https://raw.githubusercontent.com/rotki/data/main/airdrops/ens.csv',
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/airdrops.py:88]: I completely forgot we had a data repo!

File: rotkehlchen/constants/assets.py [hunk 2]
```diff
@@ -2185,3 +2185,15 @@
...
 CONSTANT_ASSETS.append(A_FOX)
+A_ENS = EthereumToken.initialize(
+    address=string_to_ethereum_address('0xC18360217D8F7Ab5e7c516566761Ea12Ce7F9D72'),
+    decimals=18,
+    name="Ethereum Name Service",
+    symbol='ENS',
+    started=Timestamp(1635800117),
+    swapped_for=None,
+    coingecko='ethereum-name-service',
+    cryptocompare=None,
```
Comment (LefterisJP) [inline rotkehlchen/constants/assets.py:2196]: It exist in CC: https://www.cryptocompare.com/coins/ens/overview

File: rotkehlchen/data/global.db.sqbpro [hunk 3]
```diff
@@ -0,0 +1,48 @@
...chen/data/global.db" readonly="0" foreign_keys="1" case_sensitive_like="0" temp_store="0" wal_autocheckpoint="1000" synchronous="2"/><attached/><window><main_tabs open="structure browser pragmas query" current="3"/></window><tab_structure><column_width id="0" width="300"/><column_width id="1" width="0"/><column_width id="2" width="100"/><column_width id="3" width="3221"/><column_width id="4" width="0"/><expanded_item id="0" parent="1"/><expanded_item id="1" parent="1"/><expanded_item id="2" parent="1"/><expanded_item id="3" parent="1"/></tab_structure><tab_browse><current_table name="4,11:mainasset_types"/><default_encoding codec=""/><browse_table_settings/></tab_browse><tab_sql><sql name="SQL 1">INSERT INTO ethereum_tokens(address, decimals, protocol) VALUES(&quot;0x2565ae0385659badCada1031DB704442E1b69982&quot;, 18, NULL); INSERT INTO assets(identifier, type, name, symbol, started, swapped_for, coingecko, cryptocompare, details_reference) VALUES(&quot;_ceth_0x2565ae0385659badCada1031DB704442E1b69982&quot;, &quot;C&quot;, &quot;ASSEMBLE&quot;, &quot;ASM&quot;, 1589092948, NULL, &quot;assemble-protocol&quot;, &quot;ASM&quot;, &quot;0x2565ae0385659badCada1031DB704442E1b69982&quot;);
```
Comment (LefterisJP) [inline rotkehlchen/data/global.db.sqbpro:1]: ehm .... wtf is this? Don't commit this file.

File: rotkehlchen/tests/unit/test_assets.py [hunk 4]
```diff
@@ -90,7 +90,7 @@ def test_cryptocompare_asset_support(cryptocompare):
         'CTX',     # Centauri coin but CarTaxi in CC
         ethaddress_to_identifier('0xf14922001A2FB8541a433905437ae954419C2439'),  # noqa: E501 # Direct insurance token but DitCoin in CC
         'DRM',     # Dreamcoin but Dreamchain in CC
-        ethaddress_to_identifier('0x82fdedfB7635441aA5A92791D001fA7388da8025'),  # noqa: E501 # Digital Ticks but Data Exchange in CC
+        ethaddress_to_identifier('0x82fdedfB7635441aA5A92791D001fA7388da8025'),  # noqa: E501 # test_cryptocompare_asset_supportDigital Ticks but Data Exchange in CC
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/test_assets.py:93]: What are you doing here? Is this a wrong copy paste?

File: rotkehlchen/data/global.db.sqbpro (hunk 3 above)
Comment (yabirgb) [reply rotkehlchen/data/global.db.sqbpro]: I don't know really

File: rotkehlchen/tests/unit/test_assets.py (hunk 4 above)
Comment (yabirgb) [reply rotkehlchen/tests/unit/test_assets.py]: Yes, from a search and I didn't notice

**Comment by yabirgb** [comment]
Not sure why that test failed. Need to check

**Comment by LefterisJP** [comment]
> Not sure why that test failed. Need to check



> Not sure why that test failed. Need to check

You need to change the recent DB info backup test as you changed global DB assets version

```
2021-11-10T08:02:39.8281714Z >       assert result['globaldb'] == {'globaldb_assets_version': 10, 'globaldb_schema_version': 2}
2021-11-10T08:02:39.8283109Z E       AssertionError: assert {'globaldb_as...a_version': 2} == {'globaldb_as...a_version': 2}
2021-11-10T08:02:39.8284314Z E         Omitting 1 identical items, use -vv to show
2021-11-10T08:02:39.8285093Z E         Differing items:
2021-11-10T08:02:39.8286124Z E         {'globaldb_assets_version': 11} != {'globaldb_assets_version': 10}
2021-11-10T08:02:39.8287406Z E         Use -v to get the full diff
2021-11-10T08:02:39.8287943Z 
2021-11-10T08:02:39.8288750Z rotkehlchen/tests/api/test_database.py:34: AssertionError
```

**Comment by yabirgb** [comment]
> > Not sure why that test failed. Need to check
> 
> > Not sure why that test failed. Need to check
> 
> You need to change the recent DB info backup test as you changed global DB assets version
> 
> ```
> 2021-11-10T08:02:39.8281714Z >       assert result['globaldb'] == {'globaldb_assets_version': 10, 'globaldb_schema_version': 2}
> 2021-11-10T08:02:39.8283109Z E       AssertionError: assert {'globaldb_as...a_version': 2} == {'globaldb_as...a_version': 2}
> 2021-11-10T08:02:39.8284314Z E         Omitting 1 identical items, use -vv to show
> 2021-11-10T08:02:39.8285093Z E         Differing items:
> 2021-11-10T08:02:39.8286124Z E         {'globaldb_assets_version': 11} != {'globaldb_assets_version': 10}
> 2021-11-10T08:02:39.8287406Z E         Use -v to get the full diff
> 2021-11-10T08:02:39.8287943Z 
> 2021-11-10T08:02:39.8288750Z rotkehlchen/tests/api/test_database.py:34: AssertionError
> ```

I was going to say that I never remembered editing that test... a git blame shows that is new :clown_face:

**Comment by yabirgb** [comment]
Okey fixed. The failing test is a timeout from coingecko that works locally

**Review by LefterisJP — APPROVED** [review]
LGTM. Will test with my addresses later and hopefully it works. Code and tests look okay.

**Pull request merged**